@xen-orchestra/fs/src/azure.js

```diff
@@ -0,0 +1,245 @@
+import { BlobServiceClient, StorageSharedKeyCredential } from '@azure/storage-blob'
+import { createLogger } from '@xen-orchestra/log'
+import { parse } from 'xo-remote-parser'
+import { join, split } from './path'
+import RemoteHandlerAbstract from './abstract'
+import { pRetry } from 'promise-toolbox'
+
+createLogger('xo:fs:azure')
+const MAX_BLOCK_SIZE = 1024 * 1024 * 1024 * 1024 * 4
+const MAX_BLOCK_COUNT = 1250
+const { warn, info } = createLogger('xo:fs:azure')
+
+export default class AzureHandler extends RemoteHandlerAbstract {
+  #container
+  #dir
+  #blobServiceClient
+  #containerClient
+
+  constructor(remote, _opts) {
+    super(remote)
+    const { username, path, password, host, protocol } = parse(remote.url)
+
+    if (host) {
+      info('Conneting to Azurite blob storage emulator...')
+      this.#blobServiceClient = BlobServiceClient.fromConnectionString(
+        `DefaultEndpointsProtocol=${protocol};AccountName=${username};AccountKey=${password};BlobEndpoint=${protocol}://${host}/${username}`
+      )
+    } else {
+      info('Conneting to Azure blob storage...')
+      const credentials = new StorageSharedKeyCredential(username, password)
+      this.#blobServiceClient = new BlobServiceClient(`${protocol}://${username}.blob.core.windows.net`, credentials)
+    }
+
+    const parts = split(path)
+    this.#container = parts.shift()
+    this.#dir = join(...parts)
+    this.#containerClient = this.#blobServiceClient.getContainerClient(this.#container)
+    const WITH_RETRY = ['_copy', '_getSize', '_list', '_outputFile', '_read', '_rename', '_unlink', '_writeFile']
+    WITH_RETRY.forEach(functionName => {
+      if (this[functionName] !== undefined) {
+        this[functionName] = pRetry.wrap(this[functionName], {
+          delays: [100, 200, 500, 1000, 2000],
+          // à réviser
+          when: err => !['BlobAlreadyExists', 'SystemInUse', 'BlobNotFound', 'ContainerNotFound'].includes(err?.code),
+          onRetry(error) {
+            warn('retrying method on fs ', {
+              method: functionName,
+              attemptNumber: this.attemptNumber,
+              delay: this.delay,
+              error,
+              file: this.arguments?.[0],
+            })
+          },
+        })
+      }
+    })
+  }
+
+  get type() {
+    return 'azure'
+  }
+
+  async _sync() {
+    await super._sync()
+    await this.#containerClient.createIfNotExists()
```

Comment:
can you test if this works without retry : 
```suggestion
    await this.#containerClient.createIfNotExists()
    await super._sync()
```